instrumentation/jmx-metrics/library/src/main/resources/jmx/rules/hadoop.yaml

```diff
@@ -0,0 +1,71 @@
+---
+rules:
+  - bean: Hadoop:service=NameNode,name=FSNamesystem
+    prefix: hadoop.dfs.
+    metricAttribute:
+      hadoop.node.name: beanattr(tag\.Hostname)
+    mapping:
+      # hadoop.dfs.capacity
+      CapacityTotal:
+        metric: capacity
+        type: updowncounter
+        unit: By
+        desc: Current raw capacity of DataNodes.
+      # hadoop.dfs.capacity.used
+      CapacityUsed:
+        metric: capacity.used
+        type: updowncounter
+        unit: By
+        desc: Current used capacity across all DataNodes.
+      # hadoop.dfs.block.count
+      BlocksTotal:
+        metric: block.count
+        type: updowncounter
+        unit: "{block}"
+        desc: Current number of allocated blocks in the system.
+      # hadoop.dfs.block.missing
+      MissingBlocks:
+        metric: block.missing
+        type: updowncounter
+        unit: "{block}"
+        desc: Current number of missing blocks.
+      # hadoop.dfs.block.corrupt
+      CorruptBlocks:
+        metric: block.corrupt
+        type: updowncounter
+        unit: "{block}"
+        desc: Current number of blocks with corrupt replicas.
+      # hadoop.dfs.volume.failure.count
+      VolumeFailuresTotal:
+        metric: volume.failure.count
+        type: counter
+        unit: "{failure}"
+        desc: Total number of volume failures across all DataNodes.
+      # hadoop.dfs.file.count
+      FilesTotal:
+        metric: file.count
+        type: updowncounter
+        unit: "{file}"
+        desc: Current number of files and directories.
+      # hadoop.dfs.connection.count
+      TotalLoad:
+        metric: connection.count
+        type: updowncounter
+        unit: "{connection}"
+        desc: Current number of connections.
+
+      # hadoop.dfs.data_node.count
+      NumLiveDataNodes:
+        metric: &metric data_node.count
+        type: &type updowncounter
+        unit: &unit "{node}"
+        desc: &desc The number of DataNodes.
+        metricAttribute:
+          hadoop.node.state: const(live)
+      NumDeadDataNodes:
+        metric: *metric
+        type: *type
+        unit: *unit
+        desc: *desc
+        metricAttribute:
+          hadoop.node.state: const(dead)
```

Comment:
I am not very familiar with hadoop, but that there are more than 2 states for the data nodes as we can infer from the `Num.*Datanodes` attributes in [doc](https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-common/Metrics.html), so I would suggest to map this as individual metrics instead of using a constant metric attribute as we can't guarantee that it's a partition.

For example, if a node would go into a state that is not mapped here, then it means that the total number of nodes in the cluster would go down by one, even if the node is still part of the cluster. With individual metrics per state, the consumer can expect that the metric does not represent the whole cluster node count.